From 2a97fdb2d344d273e07dca5ddbcdb48e3c032972 Mon Sep 17 00:00:00 2001
From: Gu Zitao <guzitao@wxiat.com>
Date: Fri, 15 Jul 2022 11:06:06 +0800
Subject: [PATCH] sw64: fix compile error and warning for CONFIG_SMP=n
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 2a97fdb2d344d273e07dca5ddbcdb48e3c032972
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/kernel/process.c | 2 +-
 arch/sw_64/kernel/time.c    | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/sw_64/kernel/process.c b/arch/sw_64/kernel/process.c
index a75ae20205f3..2508c55311ca 100644
--- a/arch/sw_64/kernel/process.c
+++ b/arch/sw_64/kernel/process.c
@@ -52,7 +52,7 @@ void arch_cpu_idle(void)
 static void common_shutdown_1(void *generic_ptr)
 {
 	struct halt_info *how = (struct halt_info *)generic_ptr;
-	int cpuid = smp_processor_id();
+	int cpuid __maybe_unused = smp_processor_id();
 
 	/* No point in taking interrupts anymore. */
 	local_irq_disable();
diff --git a/arch/sw_64/kernel/time.c b/arch/sw_64/kernel/time.c
index 15035a01e48a..0be676c80be4 100644
--- a/arch/sw_64/kernel/time.c
+++ b/arch/sw_64/kernel/time.c
@@ -4,6 +4,9 @@
 #include <linux/delay.h>
 #include <linux/time.h>
 #include <linux/clk-provider.h>
+#ifndef CONFIG_SMP
+#include <linux/clocksource.h>
+#endif
 
 #include <asm/debug.h>
 
-- 
2.33.0

